presto-main/src/main/java/com/facebook/presto/operator/aggregation/ApproximateTDigestDoublePercentileAggregations.java

```diff
@@ -0,0 +1,138 @@
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.util.Failures.checkCondition;
+import static com.google.common.base.Preconditions.checkState;
+
+@AggregationFunction("approx_percentile")
+public final class ApproximateTDigestDoublePercentileAggregations
+{
+    private static final double STANDARD_COMPRESSION_FACTOR = 100;
+
+    private ApproximateTDigestDoublePercentileAggregations() {}
+
+    @InputFunction
+    public static void input(@AggregationState TDigestAndPercentileState state, @SqlType(StandardTypes.DOUBLE) double value, @SqlType(StandardTypes.DOUBLE) double percentile)
+    {
+        initializePercentilesArray(state, createPercentileBlock(percentile));
+        initializeDigest(state, STANDARD_COMPRESSION_FACTOR);
+
+        TDigest digest = state.getDigest();
+        state.addMemoryUsage(-digest.estimatedInMemorySizeInBytes());
+        digest.add(value);
+        state.addMemoryUsage(digest.estimatedInMemorySizeInBytes());
+
+        // use last percentile
+        state.setPercentiles(ImmutableList.of(percentile));
+    }
+
+    private static Block createPercentileBlock(double percentile)
+    {
+        LongArrayBlockBuilder blockBuilder = new LongArrayBlockBuilder(null, 1);
+        DOUBLE.writeDouble(blockBuilder, percentile);
+        return blockBuilder.build();
+    }
+
+    @InputFunction
+    public static void weightedInput(@AggregationState TDigestAndPercentileState state, @SqlType(StandardTypes.DOUBLE) double value, @SqlType(StandardTypes.BIGINT) long weight, @SqlType(StandardTypes.DOUBLE) double percentile)
+    {
+        checkCondition(weight > 0, INVALID_FUNCTION_ARGUMENT, "Percentile weight must be positive");
+
+        initializePercentilesArray(state, createPercentileBlock(percentile));
+        initializeDigest(state, STANDARD_COMPRESSION_FACTOR);
+
+        TDigest digest = state.getDigest();
+        state.addMemoryUsage(-digest.estimatedInMemorySizeInBytes());
+        digest.add(value, weight);
+        state.addMemoryUsage(digest.estimatedInMemorySizeInBytes());
+
+        // use last percentile
+        state.setPercentiles(ImmutableList.of(percentile));
+    }
+
+    @InputFunction
+    public static void weightedInput(@AggregationState TDigestAndPercentileState state, @SqlType(StandardTypes.DOUBLE) double value, @SqlType(StandardTypes.BIGINT) long weight, @SqlType(StandardTypes.DOUBLE) double percentile, @SqlType(StandardTypes.DOUBLE) double compression)
+    {
+        checkCondition(weight > 0, INVALID_FUNCTION_ARGUMENT, "Percentile weight must be positive");
+
+        initializePercentilesArray(state, createPercentileBlock(percentile));
+        initializeDigest(state, compression);
+
+        TDigest digest = state.getDigest();
+        state.addMemoryUsage(-digest.estimatedInMemorySizeInBytes());
+        digest.add(value, weight);
+        state.addMemoryUsage(digest.estimatedInMemorySizeInBytes());
+
+        // use last percentile
```

Comment:
Not sure what this comment means

Reply:
Kept comment from `ApproximateLongPercentileAggregation`. It doesn't seem to have much meaning, so will delete.